src/Azure/Orleans.Transactions.AzureStorage/TransactionalState/DistributedTM/AzureTableTransactionalStateStorage.cs

```diff
@@ -0,0 +1,296 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage.Table;
+using Newtonsoft.Json;
+using Orleans.Transactions.Abstractions;
+using Orleans.Transactions.AzureStorage;
+
+namespace Orleans.Transactions.DistributedTM.AzureStorage
+{
+    public class AzureTableTransactionalStateStorage<TState> : ITransactionalStateStorage<TState>
+        where TState : class, new()
+    {
+        private readonly CloudTable table;
+        private readonly string partition;
+        private readonly JsonSerializerSettings jsonSettings;
+        private readonly ILogger logger;
+
+        private KeyEntity key;
+        private List<KeyValuePair<long, StateEntity>> states;
+
+        public AzureTableTransactionalStateStorage(CloudTable table, string partition, JsonSerializerSettings JsonSettings, ILogger<AzureTableTransactionalStateStorage<TState>> logger)
+        {
+            this.table = table;
+            this.partition = partition;
+            this.jsonSettings = JsonSettings;
+            this.logger = logger;
+        }
+
+        public async Task<TransactionalStorageLoadResponse<TState>> Load(string stateName)
```

Comment:
statename should probably be used from the interfaces calls (load, store)

Reply:
I don't prefer including statename... cleaner to specify it in the factory, which eliminates the ugly ConcurrentDictionary inside the provider.

I actually thought you already came to the same conclusion in #4538 - you removed statename from ITransactionalStateStorage, and removed the ConcurrentDictionary.

Reply:
Looks like I did not consistently update my code to reflect the latest pattern. Will do that.